### PR TITLE
Disabling ONNX IR v4 sematics for opset 8 or lower.

### PR DESCRIPTION
**Pull request by spandantiwari**
Currently, `keep_initializers_as_input` argument in `torch.onnx.export` API can be used to choose whether to export an ONNX model with IR v3 or v4 semantics. Currently, the implementation does not check for which opset is being used for export. This is an issue because ONNX IR v4 is valid only for opset 9 and above (as listed [here](https://github.com/onnx/onnx/releases/tag/v1.4.0)), and opset 8 or lower export with `keep_initializers_as_input=False` will create a illegal ONNX graph. 

This change fixes this by introducing a check on opset version when deciding whether to export ONNX IR v3 or v4.

### CODE REVIEW

File: test/onnx/test_operators.py [hunk 1]
```diff
@@ -265,6 +265,15 @@ def test_conv_onnx_irv4(self):
         x = torch.ones(20, 16, 50, 40, requires_grad=True)
         self.assertONNX(nn.Conv2d(16, 13, 3, bias=False), x)
 
+    def test_conv_onnx_irv4_opset8(self):
+        # This test point checks that for opset 8 (or lower), even if
+        # keep_initializers_as_inputs is set to False, it is ignored,
+        # and initializers are listed as ONNX graph input, in accordance
+        # with ONNX IR v3 semantics (which apply to opset version <= 8).
+        x = torch.ones(20, 16, 50, 40, requires_grad=True)
+        self.assertONNX(nn.Conv2d(16, 13, 3, bias=False), x, opset_version=8,
```
Comment (houseroad) [inline test/onnx/test_operators.py]: can we use a smaller kernel and zero them, otherwise, we may problem due to the randomness in the initializers.

Comment (spandantiwari) [reply test/onnx/test_operators.py]: Yes, that makes sense. I have updated the PR to use a smaller kernel and all ones weights. Thanks.

**Comment by spandantiwari** [comment]
@houseroad - updated the PR based on your comment.

**Comment by spandantiwari** [comment]
@houseroad - there's only one failing job, which is unrelated and seems to be a stall (`Too long with no output (exceeded 1h30m0s)`). All others have passed.

**Review by houseroad — APPROVED** [review]
Looks good, thanks.

**Pull request closed without merging**